File: assignment1/linearSearch.py
# Python code for linear searching an array
# and return the postion of the query if it exists
# else return -1, does not exist

# Search method
def search(arr, n, x):
    assert type(x) == int
    assert len(arr) < 10**3
    assert n == len(arr)
    
    for i in range (0, n):
        if arr[i] == x :
            return i;
    return -1;
    
# Driver code
def main():
    arr = [90, 22, 79, 59, 54, 12, 29, 47, 7, 96]
    n = len(arr)
    # Query
    query = input("Enter an integer query: ")
    # Searchig
    result = search(arr, n, query) 
    if result == -1: 
        print("Element is not present in array") 
    else: 
        print("Element is present at position", result+1);


# Running main module when called from source
# Using special variable
# __name__
if __name__=="__main__":
    main()










        

File: assignment1/linearSearch.cpp
// C++ code to linearly search and find
// the position of an input query value in an array
// if not present return -1

#include <iostream>

using namespace std;

int search(int arr[], int n, int x) 
{
	int i;
	for (i=0; i<n; i++) {
		if (arr[i] == x)
			return i;
	}
	return -1;
}

int main(void)
{
	int arr[] = {90, 22, 79, 59, 54, 12, 29, 47, 7, 96};
	int n = sizeof(arr) / sizeof(arr[0]);
	// input query
	int query;
	cout << "Input a query: ";
	cin >> query;
	
	int result = search(arr, n, query);
	(result == -1)? cout<<"Element is not present in array" 
		: cout<<"Element is present at position " << (result+1) << endl;
	
	return 0;
}
